modules/utils/include/hephaestus/utils/signal_handler.h

```diff
@@ -33,4 +34,58 @@ class SignalHandlerStop {
   std::atomic_flag stop_flag_ = ATOMIC_FLAG_INIT;
 };  // namespace heph::utils
 
+template <typename T>
+concept StoppableAndWaitable = requires { Stoppable<T>&& Waitable<T>; };
+
+/// \brief Use this class to block until a signal is received or the application completes.
+/// Usage:
+/// ```
+/// int main() {
+///   auto app = MyApp{};
+///   app.start().get();
+///   SignalHandlerStopOrAppComplete::wait(app);
+/// }
+/// ```
+/// > NOTE: `MayApp` needs to satify the stoppable compoenent interface.
+class SignalHandlerStopOrAppComplete {
```

Comment:
Please help me find a better name!!

Reply:
why not `StopableAndWaitableApp`?

Reply:
I asked ChatGPT and it says: `SignalAwaiter` which made me think that maybe I can just merge the two classes and provide:
```
static void wait();
template <StoppableAndWaitable T>
static void wait(T& app);
```